text/0000-process-stdio-redirection.md

```diff
@@ -0,0 +1,106 @@
+- Feature Name: process-stdio-redirection
+- Start Date: 2015-04-10
+- RFC PR:
+- Rust Issue:
+
+# Summary
+
+Update the standard library with a cross-platform high level API for redirecting
+stdio of child processes to any opened file handle or equivalent.
+
+# Motivation
+
+The current API in `std::process` allows the usage of raw file descriptors or
+HANDLEs for child process stdio redirection by leveraging the `FromRaw{Fd,
+Handle}` and `AsRaw{Fd, Handle}` trait implementations on `Stdio` and
+`ChildStd{In, Out, Err}`, respectively.
+
+Unfortunately, since the actual methods pertaining to `FromRaw*` and `AsRaw*`
+are OS specific, their usage requires either constant `cfg` checks or OS
+specific code duplication. Moreover, the conversion to `Stdio` is `unsafe` and
+requires the caller to ensure the OS handle remains valid until spawning the
```

Comment:
> However, doesn't this require that the caller mem::forget the wrapper over the descriptor/HANDLE or at least manually dup it (and perform error checking)?

I believe this is correct, yes.

> Assuming the availability of an IntoRawFd trait the currently proposed API could be tweaked to use it more appropriately:

I think one part of this that I'm uneasy about is how `Stdio::redirect(t)` is the same as `Stdio::from_raw_fd(t.into_raw_fd())`. Along those lines it seems redundant to have the `redirect` method, and then *only* adding an unsafe `redirect_by_ref` method also seems somewhat out of place because it's relatively the same thing as `Stdio::from_raw_fd(f.as_raw_fd())` + a `mem::forget` somewhere.